core/src/main/java/tc/oc/pgm/action/ActionParser.java

```diff
@@ -414,6 +418,25 @@ public FillAction parseFill(Element el, Class<?> scope) throws InvalidXMLExcepti
         parser.parseBool(el, "events").orFalse());
   }
 
+  @MethodParser("team-alias")
+  public <T extends Filterable<?>> TeamAliasAction<T> parseTeamAliasAction(
+      Element el, Class<T> scope) throws InvalidXMLException {
+    scope = parseScope(el, scope);
+    String alias = parser.string(el, "alias").required();
+
+    if (scope.getSimpleName().equals("Match")) {
+      var team = parser.reference(TeamFactory.class, el, "team").required();
+      return new TeamAliasAction.WithTeam<>(scope, alias, team);
+    } else if (scope.getSimpleName().equals("Party")) {
+      return new TeamAliasAction<>(scope, alias);
+    } else {
+      throw new InvalidXMLException(
+          "Wrong scope defined for action, scope must be " + Match.class.getSimpleName() + " or "
+              + Party.class.getSimpleName(),
+          el);
+    }
```

Comment:
```suggestion
    var action = new TeamAliasAction(alias);    
    var teamBuilder = parser.reference(TeamFactory.class, el, "team");
    var team = scope == Party.class ? teamBuilder.orNull() : teamBuilder.required();
    
    return team == null ? action : 
            new SwitchScopeAction(scope, f -> f.moduleRequire(TeamMatchModule.class).getTeam(team.get()), null, action);
```

This should make this action effectively speaking work on all scopes.
- If its party scoped, team is optional (it can still be defined, and then it will work for that team)
- If its in any other scope, team is required, and that team will be used when running the action